op-node/rollup/derive/attributes.go

```diff
@@ -116,14 +116,25 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 		upgradeTxs = append(upgradeTxs, fjord...)
 	}
 
-	l1InfoTx, err := L1InfoDepositBytes(ba.rollupCfg, sysConfig, seqNumber, l1Info, nextL2Time)
+	var hasDeposits = len(depositTxs) > 0
```

Comment:
Ah this is an interesting semantic to consider, i don't recall this being specified although its a good idea. I was originally imagining there would always be the follow up tx but this is only if there are deposits. This would imply that if there are no deposits, we call the ecotone function, and if there are deposits then we call the isthmus function

Reply:
I was not planning on it lol, do you think it's worthwhile do to the added complexity on the client to do this?
rather than "always" including the extra `isDeposit` boolean?

Reply:
I this we should just always include it for simplicity now and in the future we can consider optimizing it. So the exact semantics would be:

- if interop has activated
  - call `setL1BlockValuesIsthmus`
  - add the post deposits tx

Then we don't need to worry about conditional logic based on the number of deposits. We have this idea of deposit queue that we will likely implement in the future and we can optimize away this when we implement that